internal/http/client.go

```diff
@@ -7,20 +7,19 @@ import (
 )
 
 type Client struct {
-	client *http.Client
+	client http.Client
 }
 
-func NewHTTPClient(tlsConfig *tls.Config) (*Client, error) {
+func NewHTTPClient(tlsConfig *tls.Config) *Client {
 
-	client := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: tlsConfig,
-		},
```

Comment:
I would add here some comment here explaining why the client is created in this way.